app/controllers/subscriptions_controller.rb

```diff
@@ -29,10 +30,11 @@ def allowed_params
       [:email_on_comment, :email_on_comment_reply, :email_on_direct_message, :email_digest, :newsletter]
     end
 
-    def set_user_locale
+    def set_user_locale(&action)
       if params[:locale].blank?
-        I18n.locale = I18n.available_locales.find { |locale| locale == @user.locale&.to_sym }
-        session[:locale] = I18n.locale
+        user_locale = I18n.available_locales.find { |locale| locale == @user.locale&.to_sym }
+        session[:locale] = user_locale if user_locale.present?
```

Comment:
Hi @javierm, I added the condition just in case the line `I18n.available_locales.find { |locale| locale == @user.locale&.to_sym }` returns `nil` when the @user.locale is not defined. As I'm not sure if this case might happen I decided to add it. Another option i was thinking about:

```
    def set_user_locale(&action)
      if params[:locale].blank? && @user.locale.present?
        session[:locale] = I18n.available_locales.find { |locale| locale == @user.locale.to_sym }
      end
      I18n.with_locale(session[:locale], &action)
    end
```